js/bleutrade.js

```diff
@@ -306,6 +328,105 @@ module.exports = class bleutrade extends bittrex {
         };
     }
 
+    parseLedgerEntryType (type) {
+        // deposits don't seem to appear in here
+        const types = {
+            'TRADE': 'trade',
+            'WITHDRAW': 'withdrawal',
+        };
+        return this.safeString (types, type, type);
+    }
+
+    parseLedgerEntry (item, currency = undefined) {
+        //     trade (both sides)
+        //      { ID: 109660527,
+        //         TimeStamp: '2018-11-14 15:12:57.140776',
+        //         Asset: 'ETH',
+        //         AssetName: 'Ethereum',
+        //         Amount: 0.01,
+        //         Type: 'TRADE',
+        //         Description: 'Trade +, order id 133111123',
+        //         Comments: '',
+        //         CoinSymbol: 'ETH',
+        //         CoinName: 'Ethereum' },
+        //       { ID: 109660526,
+        //         TimeStamp: '2018-11-14 15:12:57.140776',
+        //         Asset: 'BTC',
+        //         AssetName: 'Bitcoin',
+        //         Amount: -0.00031776,
+        //         Type: 'TRADE',
+        //         Description: 'Trade -, order id 133111123, fee -0.00000079',
+        //         Comments: '',
+        //         CoinSymbol: 'BTC',
+        //         CoinName: 'Bitcoin' },
+        //
+        //    withdrawal
+        //    { ID: 104672316,
+        //     TimeStamp: '2018-05-03 08:18:19.031831',
+        //     Asset: 'DOGE',
+        //     AssetName: 'Dogecoin',
+        //     Amount: -61893.87864686,
+        //     Type: 'WITHDRAW',
+        //     Description:
+        //     'Withdraw: 61883.87864686 to address DD8tgehNNyYB2iqVazi2W1paaztgcWXtF6; fee 10.00000000',
+        //         Comments: '',
+        //     CoinSymbol: 'DOGE',
+        //     CoinName: 'Dogecoin' }
+        //
+        const currencyId = this.safeString (item, 'CoinSymbol');
+        let code = undefined;
+        if (currencyId !== undefined) {
+            code = this.commonCurrencyCode (currencyId.toUpperCase ());
+        }
+        const description = item['Description'];
+        const descParts = description.split (',');
+        const type = this.parseLedgerEntryType (this.safeString (item, 'Type'));
+        let referenceId = undefined;
+        let fee = undefined;
+        for (let i = 0; i < descParts.length; i++) {
+            const part = descParts[i].trim ();
+            if (part.startsWith ('fee')) {
+                const feeCost = this.safeFloat (part.replace ('fee -', '').trim ());
+                fee = {
+                    'cost': feeCost,
+                    'currency': code,
+                };
+            } else if (part.startsWith ('order id') && type === 'trade') {
+                referenceId = descParts[1].replace ('order id', '').trim ();
+            }
+        }
+        const time = this.safeString (item, 'TimeStamp');
+        const timestamp = this.parse8601 (time);
+        const amount = this.safeFloat (item, 'Amount');
+        let direction = undefined;
+        if (amount < 0) {
```

Comment:
We have to `Math.abs (amount)` for now, because it's documented as an absolute number.

Reply:
Ah, now I see this below.